hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java

```diff
@@ -52,10 +52,21 @@
   public Single<EntityResultSet> getEntities(EntityRequest request) {
     return this.requestBuilder
         .buildRequest(request)
-        .flatMap(serverRequest ->  this.makeRequest(request.resultSetRequest().context(), serverRequest)
-        .flatMap(serverResponse -> baselineProvider.buildRequest(serverRequest, serverResponse, request)
-        .flatMap(baselineRequest -> baselineProvider.makeRequest(request.resultSetRequest().context(), baselineRequest))
-        .flatMap(baselineResponse -> this.entityConverter.convert(request, serverResponse, baselineResponse))));
+        .flatMap(
+            serverRequest ->
```

Comment:
The formatting here is meant to expose the complexity of following this logic, which I think is now apparent. Between the comments about reducing the api of `BaselineDao`, and packaging together the responses to avoid the triconverter (not sure what the issues you ran into with that were), I think we'd see this simplify.

Reply:
Reduced the api for BaselineDao.

Reply:
I am not sure why tri-converter simplify this method,  it just reduces one argument here. Tri-converter is only used for 2 classes, i don't think it added much complexity.